src/equations/ideal_glm_mhd_1d.jl

```diff
@@ -0,0 +1,572 @@
+  rho_ll, rho_v1_ll, rho_v2_ll, rho_v3_ll, rho_e_ll, B1_ll, B2_ll, B3_ll = u_ll
+  rho_rr, rho_v1_rr, rho_v2_rr, rho_v3_rr, rho_e_rr, B1_rr, B2_rr, B3_rr = u_rr
+
+  # Calculate primitive variables
+  v1_ll = rho_v1_ll/rho_ll
+  v2_ll = rho_v2_ll/rho_ll
+  v3_ll = rho_v3_ll/rho_ll
+  vel_norm_ll = v1_ll^2 + v2_ll^2 + v3_ll^2
+  mag_norm_ll = B1_ll^2 + B2_ll^2 + B3_ll^2
+  p_ll = (equations.gamma - 1)*(rho_e_ll - 0.5*rho_ll*vel_norm_ll - 0.5*mag_norm_ll)
+
+  v1_rr = rho_v1_rr/rho_rr
+  v2_rr = rho_v2_rr/rho_rr
+  v3_rr = rho_v3_rr/rho_rr
+  vel_norm_rr = v1_rr^2 + v2_rr^2 + v3_rr^2
+  mag_norm_rr = B1_rr^2 + B2_rr^2 + B3_rr^2
+  p_rr = (equations.gamma - 1)*(rho_e_rr - 0.5*rho_rr*vel_norm_rr - 0.5*mag_norm_rr)
+
+  # compute total pressure which is thermal + magnetic pressures
+  p_total_ll = p_ll + 0.5*mag_norm_ll
+  p_total_rr = p_rr + 0.5*mag_norm_rr
+
+  # compute the Roe density averages
+  sqrt_rho_ll = sqrt(rho_ll)
+  sqrt_rho_rr = sqrt(rho_rr)
+  inv_sqrt_rho_add  = 1.0 / (sqrt_rho_ll + sqrt_rho_rr)
+  inv_sqrt_rho_prod = 1.0 / (sqrt_rho_ll * sqrt_rho_rr)
+  rho_ll_roe =  sqrt_rho_ll * inv_sqrt_rho_add
+  rho_rr_roe =  sqrt_rho_rr * inv_sqrt_rho_add
+  # Roe averages
+  # velocities and magnetic fields
+  v1_roe = v1_ll * rho_ll_roe + v1_rr * rho_rr_roe
+  v2_roe = v2_ll * rho_ll_roe + v2_rr * rho_rr_roe
+  v3_roe = v3_ll * rho_ll_roe + v3_rr * rho_rr_roe
+  B1_roe = B1_ll * rho_ll_roe + B1_rr * rho_rr_roe
+  B2_roe = B2_ll * rho_ll_roe + B2_rr * rho_rr_roe
+  B3_roe = B3_ll * rho_ll_roe + B3_rr * rho_rr_roe
+  # enthalpy
+  H_ll  = (rho_e_ll + p_total_ll) / rho_ll
+  H_rr  = (rho_e_rr + p_total_rr) / rho_rr
+  H_roe = H_ll * rho_ll_roe + H_rr * rho_rr_roe
+  # temporary vairable see equations (4.12) in Cargo and Gallice
+  X = 0.5 * ( (B1_ll - B1_rr)^2 + (B2_ll - B2_rr)^2 + (B3_ll - B3_rr)^2 ) * inv_sqrt_rho_add^2
+  # averaged components needed to compute c_f, the fast magnetoacoustic wave speed
+  b_square_roe = (B1_roe^2 + B2_roe^2 + B3_roe^2) * inv_sqrt_rho_prod # scaled magnectic sum
+  a_square_roe = ((2.0 - equations.gamma) * X +
+                 (equations.gamma -1.0) * (H_roe - 0.5*(v1_roe^2 + v2_roe^2 + v3_roe^2) -
+                                          b_square_roe)) # acoustic speed
+  # finally compute the average wave speed and set the output velocity (depends on orientation)
+  if direction == 1 # x-direction
+    c_a_roe = B1_roe^2 * inv_sqrt_rho_prod # (squared) Alfvén wave speed
+    a_star_roe = sqrt( (a_square_roe + b_square_roe)^2 - 4.0 * a_square_roe * c_a_roe )
+    c_f_roe = sqrt( 0.5 * (a_square_roe + b_square_roe + a_star_roe) )
+    vel_out_roe = v1_roe
+  else # y-direction
+    c_a_roe = B2_roe^2 * inv_sqrt_rho_prod # (squared) Alfvén wave speed
+    a_star_roe = sqrt( (a_square_roe + b_square_roe)^2 - 4.0 * a_square_roe * c_a_roe )
+    c_f_roe = sqrt( 0.5 * (a_square_roe + b_square_roe + a_star_roe) )
+    vel_out_roe = v2_roe
+  end
```

Comment:
```suggestion
  # ignore direction since it's always 1 in 1D
  c_a_roe = B1_roe^2 * inv_sqrt_rho_prod # (squared) Alfvén wave speed
  a_star_roe = sqrt( (a_square_roe + b_square_roe)^2 - 4.0 * a_square_roe * c_a_roe )
  c_f_roe = sqrt( 0.5 * (a_square_roe + b_square_roe + a_star_roe) )
  vel_out_roe = v1_roe
```

Reply:
whoops! I missed this direction if statement